mq/main/bridge/bridge-api/src/test/java/com/sun/messaging/bridge/api/StompFrameMessageTest.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2022 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0, which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception, which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ */
+
+package com.sun.messaging.bridge.api;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class StompFrameMessageTest {
+
+    /*
+     * "All frames except the CONNECT and CONNECTED frames will also escape any carriage return, line feed or colon
+     * found in the resulting UTF-8 encoded headers."
+     * - https://stomp.github.io/stomp-specification-1.2.html#Value_Encoding
+     */
+    @Test
+    void unescapeSpecialChars() {
+        // carriage return, line feed, colon and backslash
+        assertEquals("\r\n:\\", StompFrameMessage.unescapeSpecialChars("\\r\\n\\c\\\\"));
+
+        // opposite order
+        assertEquals("\\:\n\r", StompFrameMessage.unescapeSpecialChars("\\\\\\c\\n\\r"));
+
+        // real world examples
+        assertEquals("c:\\Users", StompFrameMessage.unescapeSpecialChars("c\\c\\\\Users"));
+        assertEquals("http:\\\\server:port", StompFrameMessage.unescapeSpecialChars("http\\c\\\\\\\\server\\cport"));
+
+        // backslash, c, colon ("\c:")
+        // "\\c\c" is incorrectly unescaped to "::" instead of "\c:"
```

Comment:
I'm not sure how to read this. Is there something to be changed in `StompFrameMessage`?

Reply:
Yes, I have committed a fix. The new unescapeSpecialChars passes all tests. 

I inserted a comment where illegal characters should be handled by throwing an exception, but as it is a static method the _parseException is not accessible. Please advise how the method may indicate an error. (The Specification mandates that "Undefined escape sequences such as \t (octet 92 and 116) MUST be treated as a fatal protocol error.")